### ISSUE TITLE
Tags are not stripped for promQL query result with M3-Restrict-By-Tags-JSON header specified

### ISSUE THREAD

**Issue by tsharju** [m3db/m3#3471]
When querying data using the promQL API with `M3-Restrict-By-Tags-JSON` HTTP header specified the `strip` parameter has no effect. Using the m3query endpoint instead of the prometheus one, tags are stripped. Detailed instructions on how to reproduce below.

1. What service is experiencing the issue? (M3Coordinator, M3DB, M3Aggregator, etc)

M3Coordinator

2. What is the configuration of the service? Please include any YAML files, as well as namespace / placement configuration (with any sensitive information anonymized if necessary).

Default configuration.

3. How are you using the service? For example, are you performing read/writes to the service via Prometheus, or are you using a custom script?

Reading data using the promQL API.

4. Is there a reliable way to reproduce the behavior? If so, please provide detailed instructions.

Perform e.g. following query to M3Coordinator promQL API.

```bash
curl -v -H "M3-Restrict-By-Tags-JSON: {\"match\":[{\"name\":\"cpu\",\"type\":\"EQUAL\",\"value\":\"cpu-total\"}], \"strip\": [\"host\"]}" http://localhost:17201/api/v1/query\?query\=cpu_usage_idle | jq .
```

Even though `strip` is specified the host tag is included in the result. The match filter appears to be working as it should.

```json
{
  "status": "success",
  "data": {
    "resultType": "vector",
    "result": [
      {
        "metric": {
          "__name__": "cpu_usage_idle",
          "cpu": "cpu-total",
          "host": "localhost"
        },
        "value": [
          1620278641.436,
          "87.59224515330672"
        ]
      }
    ]
  }
}
```

If I query against the m3query endpoint the `strip` appears to be working. The same seems to apply to prometheus remote read API as well. 

With this query...

```bash
curl -v -H "M3-Restrict-By-Tags-JSON: {\"match\":[{\"name\":\"cpu\",\"type\":\"EQUAL\",\"value\":\"cpu-total\"}], \"strip\": [\"host\"]}" http://localhost:17201/m3query/api/v1/query\?query\=cpu_usage_idle | jq .
```

The result is this, which seems to be correct as the host tag is stripped from the result.

```
{
  "status": "success",
  "data": {
    "resultType": "vector",
    "result": [
      {
        "metric": {
          "__name__": "cpu_usage_idle",
          "cpu": "cpu-total"
        },
        "value": [
          1620278750,
          "82.72008037192512"
        ]
      }
    ]
  }
}
```

**Comment by gibbscullen** [comment]
@tsharju Thanks for filing this issue! We have added this to our backlog.

**Comment by pahla1** [comment]
This issue is not resolved for me with M3DB v1.4.1. when I query m3query with Strip option, the tag appears in the result. any idea?
